src/ascii/interger.cairo

```diff
@@ -0,0 +1,220 @@
+trait IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252>;
+    fn to_inverse_ascii_array(self: T) -> Array<felt252>;
+}
+
+// converts intergers into an array of its individual ascii values
+// e.g. 123 -> [49, 50, 51]
+impl IntergerToAsciiArrayTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
+    impl TPartialOrd: PartialOrd<T>,
+    impl TDivRem: DivRem<T>,
+    impl TInto: Into<T, felt252>,
+    impl TryInto: TryInto<felt252, T>,
+    impl TTryIntoZero: TryInto<T, NonZero<T>>,
+    impl TZeroable: Zeroable<T>,
+    impl TDrop: Drop<T>,
+    impl TCopy: Copy<T>,
+> of IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = self.to_inverse_ascii_array();
+        new_arr.reverse();
+        new_arr
+    }
+
+    fn to_inverse_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = ArrayTrait::new();
+        if self <= 9.try_into().unwrap() {
+            new_arr.append(self.into() + 48);
+            return new_arr;
+        }
+
+        let mut num = self;
+        loop {
+            if num.is_zero() {
+                break ();
+            }
+            let (quotient, remainder) = DivRem::div_rem(
+                num, TryInto::<felt252, T>::try_into(10).unwrap().try_into().expect('Division by 0')
+            );
+            new_arr.append(remainder.into() + 48);
+            num = quotient;
+        };
+        new_arr
+    }
+}
+
+// gneric implementation for small intergers <u128 
+// to transform its intergers into a string represented as a single felt252
+// e.g. 1000 -> "1000"
+impl SmallIntergerToAsciiTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
+    impl TPartialOrd: PartialOrd<T>,
+    impl TDivRem: DivRem<T>,
+    impl TInto: Into<T, felt252>,
+    impl TTryInto: TryInto<felt252, T>,
+    impl TTryIntoZero: TryInto<T, NonZero<T>>,
+    impl TZeroable: Zeroable<T>,
+    impl TDrop: Drop<T>,
+    impl TCopy: Copy<T>,
+> of IntergerToAsciiTrait<T, felt252> {
+    fn to_ascii(self: T) -> felt252 {
+        if self <= 9.try_into().unwrap() {
+            return self.into() + 48;
+        }
+
+        let inverse_ascii_arr = self.to_inverse_ascii_array();
+        let len = inverse_ascii_arr.len();
+        let mut index = 0;
+        let mut ascii: felt252 = 0;
+        loop {
+            if index >= len {
+                break ();
+            }
+            // recursively keep getting the index from the end of the array
+            let l_index = len - index - 1;
+            ascii = ascii * 256 + *inverse_ascii_arr[l_index];
+            index += 1;
+        };
+        ascii
+    }
+}
+
+// gneric implementation for big intergers u128 
+// to transform its intergers into a string represented as multiple felt252 if there is overflow
+// e.g. max_num + 123 -> ["max_num", "123"]
+impl BigIntergerToAsciiTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
```

Comment:
is that needed ?

Reply:
i was talking about this line `impl TNumericLiteral: NumericLiteral<T>,`

Reply:
oh wait i just realise you are talking about the TNumericLiteral generic impl il remove it and test it out